### DISCUSSION TITLE
Functions and Stored procedures

### DISCUSSION THREAD

**Discussion by sudheshnakoroth** [ariga/atlas#2233] [Q&A] [answered]
Hello,

I have recently updated to the latest version of Atlas (version 15) in my development database. After this update, I created both a function and a stored procedure. However, when I used the "diff" command to generate an SQL file, it only displays the table changes and doesn't include the functions and procedures in the output.

I would greatly appreciate it if someone could provide guidance on how to generate migration scripts that include functions and procedures when using the latest version of Atlas.

Thank you for your assistance.

Best regards,

**Comment by rotemtam** [comment, marked as answer]
Hi @sudheshnakoroth 

Can you validate that you are logged in ? This feature is in beta and only available to logged in users.

**Comment by sudheshnakoroth** [comment]
@rotemtam thank you for your reply, I was not logged in was working from VS.
Do I need to purchase license to test these functionalities..
Also is there any sample for Lint and Edit commands, or is there any documentation with the sample for all the latest updates.

Thanks & Regards

**Reply by rotemtam** [reply]
We haven't added a documentation page for interactive declarative migrations yet, so the blog post is all there is currently. But essentially its just logging in and using `atlas schema apply` , when prompted choose "Lint and Edit" and your browser will popup in a review/approval screen. 

> Do I need to purchase license to test these functionalities..

You need an atlas cloud account which is free for up to 5 users.

**Reply by sudheshnakoroth** [reply]
Okay thank you will check with cloud account..

**Comment by sudheshnakoroth** [comment]
@rotemtam I have checked by logging in and it worked fine..I was able to generate migration script with both function and stored procedure, but was wondering changes made on functions or procedures will not be tracked, only create and drop will be considered.
Please confirm..

Thank you

**Reply by a8m** [reply]
Function/procedure definitions and their attributes are tracked and handled by Atlas - did you face ant issue with it?. But note, if the change to the function body is not meaningful (e.g., adding an extra space to the function body) it will be ignored by Atlas because it has no meaning in the database.

**Reply by sudheshnakoroth** [reply]
@a8m I have changed num1 * num2, to num1 + num2 but it didn't trigger the change, it said there is no change..
Following is the function
CREATE OR REPLACE FUNCTION calculate_square123(num integer)
RETURNS integer AS $$
DECLARE
    result integer;
BEGIN
    result := num + num;
    RETURN result;
END;
$$ LANGUAGE plpgsql;
